src/Swarm/TUI/View.hs

```diff
@@ -511,6 +514,25 @@ descriptionTitle e = " " ++ from @Text (e ^. entityName) ++ " "
 descriptionWidget :: AppState -> Entity -> Widget Name
 descriptionWidget s e = padLeftRight 1 (explainEntry s e)
 
+messagesWidget :: GameState -> Widget Name
+messagesWidget gs = viewport MessageViewport Vertical (vBox widgetList)
+ where
+  widgetList = focusNewest . map drawLogEntry' . toList . Seq.sort $ focusedLogs <> messages
```

Comment:
This seems to fix the scroll issue, not sure why though :)
```suggestion
  widgetList = map drawLogEntry' . toList . Seq.sort $ focusedLogs <> messages
```

Reply:
It doesn't like the scroll is working properly yet.

Reply:
Fixed :white_check_mark: